src/atlas/projection/detail/StretchProjection.cc

```diff
@@ -0,0 +1,373 @@
+/**
+* (C) Crown copyright 2021, Met Office
+*
+* This software is licensed under the terms of the Apache Licence Version 2.0
+* which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+*/
+
+#include <cmath>
+#include <functional>
+#include <sstream>
+
+#include "eckit/config/Parametrisation.h"
+#include "eckit/utils/Hash.h"
+
+#include "atlas/projection/detail/StretchProjection.h"
+#include "atlas/projection/detail/ProjectionFactory.h"
+#include "atlas/runtime/Exception.h"
+#include "atlas/util/Config.h"
+#include "atlas/util/Constants.h"
+#include "atlas/grid.h"
+#include "atlas/grid/Grid.h"
+#include "atlas/functionspace/StructuredColumns.h"
+#include "eckit/testing/Test.h"
+
+/**
+* Projection for LAM stretching
+*
```

Comment:
I think that you need to have a better description of the model here with a reference to Terry Davies paper that uses it.

Reply:
/**
* Projection for LAM stretching
*
* The theory used in this code can be found under:
* The benefits of the Met Office variable resolution NWP model for forecasting convection
* Tang et al. 2013
* https://doi.org/10.1002/met.1300
*
* Generally: From a original regular grid the point in the middle of the grid are unchanged,
* based on the boundary of the new internal regular grid.
* The grid length is then stretched at a constant stretching (or inflation) factor r,
* that is Δsi+1 = rΔsi until all the points are mapped in the new grid.
* The domain is then extended to a further uniform coarse resolution region (rim) in the outer domain,
*
* The origin of the xy-system is at (lon,lat) = (0,0)
*
*/